File: parciales/ED1-Parcial2-2021-1--002/codigo/CircularQueue.py
import numpy as np
class CircularQueue:
    def __init__(self, size): 
        self.queue = np.zeros(size)
        self.front = 0
        self.rear = 0
        self.size = size
        self.currentSize = 0
    
    def add(self, x):
        if self.currentSize == self.size: 
            raise Exception("The queue is full: front size: " + str(self.rear)) 
        self.queue[self.rear] = x
        self.rear += 1
        self.rear = (self.rear + 1) % self.size #....
        self.currentSize += 1
        return True
    
    def peek(self):
        if self.front == self.rear: 
            raise Exception("The queue is empty");
        return self.queue[self.front]
    
    def poll(self):
        if self.front > self.rear: 
            raise Exception("The queue is empty");
        self.currentSize -= 1
        self.front += 1
        return self.queue[self.front-1]
    
def __main__():
        q = CircularQueue(10)
        q.add(2)
        q.add(3)
        print(q.poll())

__main__()

File: parciales/ED1-Parcial2-2021-1--002/codigo/CircularQueue.java
public class CircularQueue {
    int[] queue; 
    int front;
    int rear; 
    int currentSize;
    int size;

    public CircularQueue(int size) { 
        this.queue = new int[size];
        this.front = 0;
        this.rear = 0;
        this.size = size;
        this.currentSize = 0;
    }
    public boolean add(int x) {
        if (currentSize == size) 
            throw new IllegalStateException("The queue is full: front size: " + rear); 
        queue[rear++] = x;
        rear = (rear + 1) % size;
        currentSize++;
        return true;
    }
    public int peek() {
        if (front == rear) 
            throw new IllegalStateException("The queue is empty");
        return queue[front];
    }
    public int poll() {
        if (front > rear) 
            throw new IllegalStateException("The queue is empty");
        currentSize--;
        return queue[front++];
    }
    public static void main(String[] args){
      CircularQueue c = new CircularQueue(10);
    }   
}
